spec/src/main/asciidoc/entity.asciidoc

```diff
@@ -0,0 +1,528 @@
+----
+@Embeddable
+public class Address {
+    private String street;
+    private String city;
+    private String postalCode;
+}
+
+@Entity
+public class Person {
+    @Id
+    private Long id;
+    private String name;
+    private Address address;  // embedded field
+}
+----
+
+In a document, wide-column, or graph database, the JSON representation of an instance of the `Person` entity might be as follows:
+
+[source,json]
+----
+{
+  "id": 1,
+  "name": "John Doe",
+  "street": "123 Main St",
+  "city": "Sampleville",
+  "postalCode": "12345"
+}
+----
+
+Or, in a relational database, the DDL for the `Person` table might look like this:
+
+[source,sql]
+----
+create table Person (
+    id bigint primary key,
+    name varchar,
+    street varchar,
+    city varchar,
+    postalCode varchar
+)
+----
+
+In a structured representation, the fields of the embeddable class are somehow grouped together in the data schema.
+
+For example, the JSON representation of `Person` might be:
+
+[source,json]
+----
+{
+  "id": 1,
+  "name": "John Doe",
+  "address":
+  {
+    "street": "123 Main St",
+    "city": "Sampleville",
+    "postalCode": "12345"
+  }
+}
+----
+
+Or the SQL DDL could be:
+
+[source,sql]
+----
+create type Address as (
+    street varchar,
+    city varchar,
+    postalCode varchar
+)
+
+create table Person (
+    id bigint primary key,
+    name varchar,
+    address Address
+)
+----
+
+NOTE: Support for embeddable classes and embedded fields is not required by this specification.
+However, every Jakarta Data provider is strongly encouraged to provide support for embeddable classes within its entity programming model.
+
+==== Entity Associations
+
+An association field is a field of an entity class whose declared type is also an entity class.
+Given an instance of the first entity class, its association field holds a reference to an instance of a second entity class.
+
+For example, consider the following Java classes:
+
+[source,java]
+----
+@Entity
+public class Author {
+    @Id
+    private UUID id;
+    private String name;
+    private List<Book> books;
+}
+
+@Entity
+public class Book {
+    @Id
+    private Long id;
+    private String title;
+    private String category;
+    private List<Author> authors;
+}
+----
+
+In a relational database, these entities might map to the following data schema:
+
+[source,sql]
+----
+create table Author (
+    uuid id primary key,
+    name varchar,
+)
+
+create table BookAuthor(
+    book bigint,
+    author uuid,
+    primary key (book, author),
+    foreign key (author) references Author,
+    foreign key (book) references Book
+)
+
+create table Book (
+    id bigint primary key,
+    title varchar,
+    category varchar
+)
+----
+
+NOTE: Support for entity associations is not required by this specification.
+
+=== Entity Names and Persistent Field Names
+
+Entities and their persistent fields may be referenced by name in the query language defined in <<Jakarta Data Query Language>>.
```

Comment:
Note that this one-sentence preamble to section 3.2 is the only change to the actual words.